sql/core/src/test/scala/org/apache/spark/status/protobuf/sql/KVStoreProtobufSerializerSuite.scala

```diff
@@ -0,0 +1,88 @@
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.execution.ui.SQLExecutionUIData
+import org.apache.spark.status.api.v1.sql.SqlResourceSuite
+import org.apache.spark.status.protobuf.KVStoreProtobufSerializer
+
+class KVStoreProtobufSerializerSuite extends SparkFunSuite {
+
+  private val serializer = new KVStoreProtobufSerializer()
+
+  test("SQLExecutionUIData") {
+    val input = SqlResourceSuite.sqlExecutionUIData
+    val bytes = serializer.serialize(input)
+    val result = serializer.deserialize(bytes, classOf[SQLExecutionUIData])
+    assert(result.executionId == input.executionId)
+    assert(result.description == input.description)
+    assert(result.details == input.details)
+    assert(result.physicalPlanDescription == input.physicalPlanDescription)
+    assert(result.modifiedConfigs == input.modifiedConfigs)
+    assert(result.metrics == input.metrics)
+    assert(result.submissionTime == input.submissionTime)
+    assert(result.completionTime == input.completionTime)
+    assert(result.errorMessage == input.errorMessage)
+    assert(result.jobs == input.jobs)
+    assert(result.stages == input.stages)
+    assert(result.metricValues == input.metricValues)
+  }
+
+  test("SQLExecutionUIData with metricValues is empty map and null") {
+    val templateData = SqlResourceSuite.sqlExecutionUIData
+
+    val input1 = new SQLExecutionUIData(
+      executionId = templateData.executionId,
+      description = templateData.description,
+      details = templateData.details,
+      physicalPlanDescription = templateData.physicalPlanDescription,
+      modifiedConfigs = templateData.modifiedConfigs,
+      metrics = templateData.metrics,
+      submissionTime = templateData.submissionTime,
+      completionTime = templateData.completionTime,
+      errorMessage = templateData.errorMessage,
+      jobs = templateData.jobs,
+      stages = templateData.stages,
+      metricValues = Map.empty
+    )
+    val bytes1 = serializer.serialize(input1)
+    val result1 = serializer.deserialize(bytes1, classOf[SQLExecutionUIData])
+    // input.metricValues is empty map, result.metricValues is empty map.
+    assert(result1.metricValues.isEmpty)
+
+    val input2 = new SQLExecutionUIData(
+      executionId = templateData.executionId,
+      description = templateData.description,
+      details = templateData.details,
+      physicalPlanDescription = templateData.physicalPlanDescription,
+      modifiedConfigs = templateData.modifiedConfigs,
+      metrics = templateData.metrics,
+      submissionTime = templateData.submissionTime,
+      completionTime = templateData.completionTime,
+      errorMessage = templateData.errorMessage,
+      jobs = templateData.jobs,
+      stages = templateData.stages,
+      metricValues = null
+    )
+    val bytes2 = serializer.serialize(input2)
+    val result2 = serializer.deserialize(bytes2, classOf[SQLExecutionUIData])
+    // input.metricValues is null, result.metricValues is also empty map.
```

Comment:
@gengliangwang For the collection type, the input is null, and after deserialization is empty collection, which may be different from json

Reply:
Returning an empty collection seems safer. It won't affect the UI page anyway. Thanks for noticing it.

Reply:
OK